src/cdk-pipelines/slack-notification.ts

```diff
@@ -103,9 +100,11 @@ export class SlackNotification extends constructs.Construct {
       }),
     )
 
+    props.slackWebhookUrlSecret.grantRead(reportFunction)
+
     props.artifactsBucket.grantRead(reportFunction, props.triggerObjectKey)
 
-    props.pipeline.onStateChange("Event" + (props.singletonLambdaUuid ?? ""), {
+    props.pipeline.onStateChange("Event" + (props.id ?? "1"), {
```

Comment:
Is there a reason for introducing a separate `id` property and not using the construct id that we already have?
```suggestion
    props.pipeline.onStateChange("Event${id}`), {
```

Reply:
Nei, jeg tenkte bare ikke på det! Gjør det sånn i stedet